src/apps/sorting-visualizer/components/controller/array-input.tsx

```diff
@@ -1,56 +1,98 @@
-import {
-  convertArrayStringToArray,
-  convertInputToArrayString,
-  getRndmNumInRange,
-} from "@/apps/sorting-visualizer/helpers/array-helpers";
+import React, { useState, useEffect } from 'react'
 import {
   setArray,
   setIsPlaying,
-  setReset,
-} from "@/apps/sorting-visualizer/store/sorting-visualizer.slice";
-import { useAppDispatch, useAppSelector } from "@/store/hooks";
-import { useEffect, useState } from "react";
+  setReset
+} from '@/apps/sorting-visualizer/store/sorting-visualizer.slice'
+import { useAppDispatch, useAppSelector } from '@/store/hooks'
+import {
+  convertArrayStringToArray,
+  convertInputToArrayString,
+  getRndmNumInRange
+} from '@/apps/sorting-visualizer/helpers/array-helpers'
 
-import classes from "./controls.module.scss";
+import classes from './controls.module.scss'
 
-function ArrayInput() {
-  const dispatch = useAppDispatch();
-  const array = useAppSelector((state) => state.sortViz.array);
-  const [input, setInput] = useState(array.join(", "));
+type SortFunction = () => void
+
+function ArrayInput () {
+  const dispatch = useAppDispatch()
+  const array = useAppSelector(state => state.sortViz.array)
+  const [input, setInput] = useState(array.join(', '))
+  // const [sortingOptionsVisible, setSortingOptionsVisible] = useState(false)
 
   useEffect(() => {
-    dispatch(setIsPlaying(false));
-    dispatch(setReset());
-  }, [array, dispatch]);
+    dispatch(setIsPlaying(false))
+    dispatch(setReset())
+  }, [array, dispatch])
 
   const onRandomize = () => {
     const newInput = Array.from(new Array(getRndmNumInRange(10, 40)), () =>
       getRndmNumInRange()
-    );
-    setInput(newInput.join(", "));
-    dispatch(setArray(newInput));
-  };
+    )
+    setInput(newInput.join(', '))
+    dispatch(setArray(newInput))
+  }
+
+  const handleSortAscending = () => {
+    const sortedAscending: number[] = input
+      .split(', ')
+      .map(num => parseInt(num, 10))
+      .sort((a, b) => a - b)
+    setInput(sortedAscending.join(', '))
+    dispatch(setArray(sortedAscending))
+  }
+
+  const handleSortDescending = () => {
+    const sortedDescending: number[] = input
+      .split(', ')
+      .map(num => parseInt(num, 10))
+      .sort((a, b) => b - a)
+    setInput(sortedDescending.join(', '))
+    dispatch(setArray(sortedDescending))
+  }
```

Comment:
onRandomize, handleSortAscending, handleSortDescending can be moved to helpers and keep the component lean